tests/test_resources/test_envs/test_env.py

```diff
@@ -343,25 +37,168 @@ def np_summer(a, b):
     return int(np.sum([a, b]))
 
 
-@pytest.mark.clustertest
-def test_conda_call_fn(cluster):
-    conda_dict = _get_conda_env(name="c")
-    conda_env = rh.env(conda_env=conda_dict, reqs=["pytest", "numpy"])
-    fn = rh.function(np_summer).to(system=cluster, env=conda_env)
-    result = fn(1, 4)
-    assert result == 5
-
+class TestEnv(TestResource):
 
-@unittest.skip("Map does not work properly following Module refactor.")
-@pytest.mark.clustertest
-def test_conda_map_fn(cluster):
-    conda_dict = _get_conda_env(name="test-map-fn")
-    conda_env = rh.env(conda_env=conda_dict, reqs=["pytest", "numpy"])
-    map_fn = rh.function(np_summer, system=cluster, env=conda_env)
-    inputs = list(zip(range(5), range(4, 9)))
-    results = map_fn.starmap(inputs)
-    assert results == [4, 6, 8, 10, 12]
+    MAP_FIXTURES = {"resource": "env"}
 
+    UNIT = {
+        "env": [
+            base_env,
+            base_conda_env,
+            conda_env_from_dict,
+            conda_env_from_local,
+            conda_env_from_path,
+        ]
+    }
+    LOCAL = {
+        "env": [
+            base_env,
+            base_conda_env,
+            conda_env_from_dict,
+            conda_env_from_local,
+            conda_env_from_path,
+        ],
+        # TODO: add local clusters once conda docker container is set up
+    }
+    MINIMAL = {"env": [base_env, base_conda_env], "cluster": [ondemand_cpu_cluster]}
+    THOROUGH = {
+        "env": [
+            base_env,
+            base_conda_env,
+            conda_env_from_dict,
+            conda_env_from_local,
+            conda_env_from_path,
+        ],
+        "cluster": [ondemand_cpu_cluster, password_cluster],
+    }
+    MAXIMAL = {
+        "env": [
+            base_env,
+            base_conda_env,
+            conda_env_from_dict,
+            conda_env_from_local,
+            conda_env_from_path,
+        ],
+        "cluster": [ondemand_cpu_cluster, password_cluster],
+    }
 
-if __name__ == "__main__":
-    unittest.main()
+    @pytest.mark.level("unit")
+    def test_env_factory_and_properties(self, env):
+        assert isinstance(env, rh.Env)
+
+        args = init_args[id(env)]
+        if "reqs" in args:
+            assert set(args["reqs"]).issubset(env.reqs)
+
+        if isinstance(env, rh.CondaEnv):
+            assert env.conda_yaml
+
+    @pytest.mark.level("minimal")
+    def test_env_to_cluster(self, env, cluster):
+        env.to(cluster)
+
+        for req in env.reqs:
+            if not req == "./":
+                res = cluster.run([f"pip freeze | grep {req}"], env=env)
+                assert res[0][0] == 0  # installed properly
+
+        _uninstall_env(env, cluster)
+
+    @unittest.skip("Running into s3 folder issue")
```

Comment:
test currently failing, from trying to copy rh.folder to s3. sky added a `_assert_kwargs_builtin_type` check recently that's resulting `AssertionError: kwargs should not contain none built-in types: {'region_name': None}`. needs to be addressed for folder tests to pass as well.